src/geometry.rs

```diff
@@ -209,6 +219,15 @@ impl TriangleFace {
             normals: (ni1, ni2, ni3),
         }
     }
+
+    /// Generates 3 edges from the respective triangular face
+    pub fn to_edges(&self) -> Vec<Edge> {
```

Comment:
Instead of `Vec<Edge>` I would make the return type an `[Edge; 3]`. You prevent a heap allocation and get more static checks for free.

Reply:
can you then `flat_map` `[Edge; 3]`?

Reply:
Yes, `flat_map` is `Iterator` method, so you'll have to convert array into iterator through `.into_iter()`.